### ISSUE TITLE
Expose Cluster Operations Events/Logs

### ISSUE THREAD

**Issue by vlerenc** [gardener/dashboard#26]
# Story
- As operator (of the week) I want to access cluster operations logs (create, update/reconcile, delete) in a convenient way, so that I can easily see what the outcome of each operation was without trying to `grep` the information out of the overall/global Gardener logs.

# Motivation
We like to improve the way how we interact/access the Gardener/cluster logs. As an operator (of the week) I frequently have to check what the Gardener says/logs on a certain cluster and operation. Now with the reconciler, the logs grew and with all the other planned features, they will even grow more. Of course, we will have a logging stack eventually, but that's not what this story is about. Usually, we need to know something specific about a particular cluster, for a particular operation at a particular date/time. A logging stack and UI will allow me to set filters, but it won't be semantic in this sense. Also, we can't give away the entire operator logs via a logging stack to the entirety of the people that will later use the Gardener dashboard as operators (there must be strong isolation, so that we can have only one Gardener instance running). Therefore the Gardener keeps track of all operations with an operation ID and logs all operations with their ID as events at the shoot cluster resource.

# Acceptance Criteria
- [ ] Show Gardener operations (and their status/results) per cluster (CRD events)
- [ ] Show logs per Gardener operation (fetched from the Gardener logs for now)

# Definition of Done
- [ ] Knowledge is distributed: Have you spread your knowledge in pair programming/code review?
- [ ] Unit Tests are provided: Have you written automated unit tests or added manual NGPTT tickets?
- [ ] Integration Tests are provided: Have you written automated integration tests?
- [ ] Minimum API exposure: If you have added public API, was it really necessary/is it minimal?
- [ ] Operations guide: Have you updated the [operations guide](https://github.wdf.sap.corp/kubernetes/kube-docs/wiki/Operations%20Guide)?

**Comment by grolu** [comment]
We will leverage the terminal feature to achieve this. Also expose cluster insights like deployments, worker, pods, etc.

**Comment by grolu** [comment]
See also #8 - We want to allow the user to configure custom views to show custom resources, views etc. via the terminal feature on the details screen.

**Comment by grolu** [comment]
Maybe also part of #26. Let's see how far we get by leveraging the terminal feature

**Comment by vlerenc** [comment]
Isn't here the main problem, how to get to the cluster operation logs in the first place @grolu , cc @rfranzke?

**Comment by grolu** [comment]
Yes. And actually I moved all issues of this kind to in review. As in my point of view the terminal shortcuts feature (https://github.com/gardener/dashboard/pull/739) can be at least a temporary way to solve this @petersutter

**Comment by petersutter** [comment]
Well it depends on how this information is fetched / from where. @rfranzke is there an example anywhere?

**Comment by rfranzke** [comment]
Can you explain again? What do you want to fetch? What are cluster operation logs?

**Comment by vlerenc** [comment]
I think, this is a misunderstanding. This is a very old story or rather epic and it is most likely nothing that can be "just implemented". It will need planning and possibly more side changes.

So let's first get back to the goal:
- Show Gardener operations (and their status/results) per cluster (shoot resource event log)

That can be a creation, upgrade, reconciliation/maintenance operation. So the first question would be whether that's possible. Second how. Today, we have no longer GRM doing everything. Initiation happens there, but the rest is happening in the `gardenlet`, but that aside:
- Is it useful to present this information to the end users (helpful)? The assumption is yes. End users were also asking for that?
- Are we sure, it doesn't contain sensitive information?
- If not, what do we have to do in terms of cleanup?
- Can we find it easily per cluster (by operation ID or something)?
- How can we extract this data so that we can show it to end users?
- If we can't, can we show a minimal log, e.g. just an overview which operations were executed and their result w/o any logs?

The log viewer idea (for the detailed logs) in the description is somewhat troubled, because end users should not be able to see shared resource logs like those of GRM or `gardenlet`, so is there another way?

You see, many open questions, because we haven't designed that into the solution from the beginning. We talked however about it and have done something with events, but then ran into TTL issues/discussed the matter, but I don't think conclusively, right?
